spec/factories/course_condition_achievements.rb

```diff
@@ -3,5 +3,22 @@
           class: Course::Condition::Achievement.name, aliases: [:achievement_condition] do
     course
     achievement
+    association :conditional, factory: :course_achievement
+
+    factory :self_referential_course_condition_achievement do
+      after(:build) do |achievement_condition, evaluator|
```

Comment:
Unused block argument - `evaluator`. If it's necessary, use `_` or `_evaluator` as an argument name to indicate that it won't be used.